### PR TITLE
[17.05] Fix tool id encoding

### PR DESCRIPTION
**Pull request by guerler**
Since 17.01 we use the backbone router and push state functionality more consequently. In the case of tool urls the tool id is automatically escaped. My assumption is that this can lead to problems with the underlying API requests which previously expected unescaped tool ids. If this restores the previous behavior we can prepare a follow-up PR and escape all tool ids used in other urls.

Ping @dannon. Can you take a look?

### CODE REVIEW

**Comment by martenson** [comment]
To provide some context, this is an issue raised at dev-list: http://dev.list.galaxyproject.org/new-installation-galaxy-17-05-Uncaught-error-td4670819.html

a quick deployer workaround is to add 

`AllowEncodedSlashes NoDecode`

to your Apache config, see https://galaxyproject.org/admin/config/apache-proxy/#allow-encoded-slashes-in-urls

But we believe the proper way is to not call URLs with encoded slashes (no problem if the parameters have encoded slashes).

I also think this might resolve the data_manager issue too.

**Comment by martenson** [comment]
ping @FredericBGA - could you please test this?

File: client/galaxy/scripts/apps/analysis.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ window.app = function app( options, bootstrapped ){
         /** load the center panel with a tool form described by the given params obj */
         _loadToolForm : function( params ){
             //TODO: load tool form code async
-            params.id = params.tool_id;
+            params.id = unescape( params.tool_id );
```
Comment (nsoranzo) [inline client/galaxy/scripts/apps/analysis.js:193]: @erasche Is this safe to do? Or should we just unescape `/`?

Comment (guerler) [reply client/galaxy/scripts/apps/analysis.js:193]: @erasche said its fine.

**Comment by hexylena** [comment]
I can't think of any issues off the top of my head. :+1:

**Comment by FredericBGA** [comment]
I’ve updated client/galaxy/scripts/apps/analysis.js, restart both galaxy and apache (without AllowEncodedSlashes NoDecode directive)

URLs of tools from toolshed look like :
/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_rpstblastn_wrapper%2F0.1.08&version=0.1.08&__identifer=p5d97lvw1

And it doesn’t work (“Uncaught error”)

I still need the use of AllowEncodedSlashes NoDecode



De : Martin Cech [mailto:notifications@github.com]
Envoyé : lundi 26 juin 2017 12:18
À : galaxyproject/galaxy <galaxy@noreply.github.com>
Cc : SAPET, Frederic <frederic.sapet@biogemma.com>; Mention <mention@noreply.github.com>
Objet : Re: [galaxyproject/galaxy] [17.05] Fix tool id encoding (#4225)


ping @FredericBGA<https://github.com/fredericbga> - could you please test this?

**Comment by guerler** [comment]
Thanks for testing this @FredericBGA. I will set up the Apache server later and further debug this.

**Comment by martenson** [comment]
@FredericBGA did you pull in this whole PR or just the `client/.../analysis.js`? Because you also need the build artifacts in this PR to test this change. The one file is not enough.

**Comment by FredericBGA** [comment]
@martenson you're right, I've only update the javasript file, I will test soon with a full rebuild.

**Comment by bgruening** [comment]
ping @FredericBGA, could you please test this. Thanks!

**Comment by FredericBGA** [comment]
@bgruening : Sorry for the delay! But it looks like I need to learn of bit more of Git (in order to be sure that I check with the right files...)

**Comment by FredericBGA** [comment]
@bgruening 

Ok, this PR works, 'AllowEncodedSlashes NoDecode' is no more needed with Apache.

Just to be sure, here is what I 've done:
```
git fetch origin refs/pull/4225/head:pr-4225
git checkout pr-4225
#to be sure I work with the good file
git diff release_17.05  static/scripts/galaxy.js
```

Then restart both Apache and Galaxy.

**Comment by dannon** [comment]
@guerler Sorry I missed this, that mention in the edit did not notify me.  I'm checking it out now.

**Comment by dannon** [comment]
@guerler Yeah, this looks good to me.  Merging manually to resolve build conflict and will merge forward to -dev.

**Pull request merged**